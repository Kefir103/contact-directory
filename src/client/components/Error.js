import React from 'react';

export const Error = () => {
    return (
        <div className={'error-container'}>
            <h1>Упс! :(</h1>
            <h2>Произошла ошибка!</h2>
            <h3>Пожалуйста проверьте свое интернет-соединение или повторите попытку позже</h3>
        </div>
    );
};
