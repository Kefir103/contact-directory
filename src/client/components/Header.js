import React from 'react';

export default function Header() {
    return (<header><h1>Контактный справочник</h1></header>)
}
