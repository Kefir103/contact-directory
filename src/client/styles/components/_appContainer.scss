.app-container {
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 100%;
    margin-top: 95px;
    margin-left: 250px;
    padding: 0 2%;
}
