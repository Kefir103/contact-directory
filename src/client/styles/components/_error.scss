.error-container {
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 100%;
    margin-top: 95px;

    h1 {
        font-size: 4em;
        margin: 0;
    }

    h2 {
        margin: 0;
    }

    h3 {
        text-align: center;
    }
}
