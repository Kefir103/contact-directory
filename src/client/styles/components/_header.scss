header {
    position: fixed;
    width: 100%;
    height: 80px;
    box-shadow: 2px 0 5px 2px silver;
    background: #322fb9;
    padding-left: 15px;
    color: white;
    top: 0;
    left: 0;
    right: 0;
    z-index: 1;
}
