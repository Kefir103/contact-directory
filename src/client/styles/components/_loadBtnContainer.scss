.load-btn-container {
    display: flex;
    justify-content: space-around;
    margin: 15px auto;
    width: 80%;

    .load-btn {
        height: 60px;
        border: 2px solid dimgray;
        border-radius: 5px;
        outline: none;
        background-color: lightgray;
        font-size: 14pt;

        &:hover {
            cursor: pointer;
        }
    }
}
