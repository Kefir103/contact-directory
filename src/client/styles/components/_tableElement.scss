.table-cell {
    display: flex;
    align-items: center;
    border-left: 1px solid darkgray;
    border-bottom: 1px solid darkgray;
    padding: 0 5px;
    width: 25%;
    box-sizing: border-box;
    overflow: hidden;
    &.id {
        width: 5%;
    }
    &:last-child {
        border-right: 1px solid darkgray;
    }
}
