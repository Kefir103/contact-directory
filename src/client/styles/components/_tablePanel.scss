.panel {
    height: 50px;

    .table-cell-button {
        display: flex;
        align-items: center;
        border: 2px solid dimgray;
        border-right: none;
        padding: 0 5px;
        outline: none;
        font-family: -apple-system, Arial, Helvetica, sans-serif;
        font-size: 14pt;
        box-sizing: border-box;
        width: 25%;
        &.id {
            width: 5%;
        }
        &:hover {
            cursor: pointer;
        }
        &.up-arrow:after {
            content: '\25B2';
            margin-left: auto;
        }
        &.down-arrow:after {
            content: '\25BC';
            margin-left: auto;
        }
        &:last-child {
            border-right: 2px solid dimgray;
        }
    }
}
