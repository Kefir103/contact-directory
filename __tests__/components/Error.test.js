import React from 'react';
import { Error } from '../../src/client/components/Error';
import { render } from 'enzyme';

describe('<Error />', () => {
    test('should render correctly', () => {
        const component = render(<Error />);
        expect(component).toMatchSnapshot();
    });
});
