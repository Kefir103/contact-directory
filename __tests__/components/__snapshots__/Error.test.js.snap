// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> should render correctly 1`] = `
<div
  class="error-container"
>
  <h1>
    Упс! :(
  </h1>
  <h2>
    Произошла ошибка!
  </h2>
  <h3>
    Пожалуйста проверьте свое интернет-соединение или повторите попытку позже
  </h3>
</div>
`;
